frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.tsx

```diff
@@ -138,13 +145,13 @@ export function SmartScalar(props) {
   const column = cols[metricIndex];
 
   const insights = rawSeries?.[0].data?.insights;
-  const insight = _.findWhere(insights, { col: column.name });
+  const insight = insights && _.findWhere(insights, { col: column.name });
   if (!insight) {
     return null;
   }
 
   const lastValue = insight["last-value"];
-  const formatOptions = settings.column(column);
+  const formatOptions = settings.column?.(column) as OptionsType;
```

Comment:
I think we messed up when typing `settings.column`'s return value, it seems like we defined it as something completely different than it actually is

```typescript
export type ComputedVisualizationSettings = VisualizationSettings & {
  column?: (
    col: RemappingHydratedDatasetColumn,
  ) => RemappingHydratedDatasetColumn;
};
```

@shaunlebron I think it's fine to keep your typecast here since you seen to have the right type (although it seems `OptionsType` is missing a few properties like `currency` that are present on `formatOptions`, so it might be worth adding those.

@alxnddr Let's try to fix the return type of the `settings.column` method.

Reply:
Oh yes, the return type is incorrect. It is also not `OptionsType` but `ColumnSettings` which is probably missing.
The `OptionsType` is a part of the formatting code interface